src/Microsoft.Identity.Client/TokenCache.cs

```diff
@@ -795,78 +796,152 @@ private string GetPreferredEnvironmentHost(string environmentHost, InstanceDisco
             return preferredEnvironmentHost;
         }
 
-        IEnumerable<IAccount> ITokenCacheInternal.GetAccounts(string authority)
+        async Task<IEnumerable<IAccount>> ITokenCacheInternal.GetAccountsAsync(string authority)
+        {
+            var environment = Authority.GetEnviroment(authority);
+
+            FetchAllAccountItemsFromCache(
+                environment,
+                out IEnumerable<MsalRefreshTokenCacheItem> rtCacheItems,
+                out IEnumerable<MsalAccountCacheItem> accountCacheItems,
+                out AdalUsersForMsalResult adalUsersResult);
+
+            // Multi-cloud support - must filter by env, namely by preferred_cache_env
+            var cacheEnv = await GetPreferedCacheEnvAvoidNetworkCallAsync(authority, accountCacheItems).ConfigureAwait(false);
```

Comment:
>Prefered [](start = 36, length = 8)

nit: spelling Preferred

Reply:
Yes, I also prefer to have proper spelling 👍